grr/server/grr_response_server/gui/ui/components/client_details/client_diff.ts

```diff
@@ -205,51 +205,67 @@ export function getClientVersions(clientSnapshots: Client[]):
 
 function getFirstStringsJoinedPath(path: any[]): string {
   let tokens: string[] = [];
-  path.forEach(val => {
-    if (typeof val !== 'string') return;
-    tokens.push(val);
-  });
+
+  for (let i = 0; i < path.length; i++) {
+    if (typeof path[i] !== 'string') break;
+    tokens.push(path[i]);
+  }
```

Comment:
Please explain why you stop appending after the first non-string. What is a typical non-string value here? Why are strings after the non-string not relevant?

Reply:
A typical non-string element would be a number [0,1,..n] for array access, because `deep-diff` can find differences between individual elements of an array. Unfortunately, showing detailed versions of array individual elements is not currently supported by us in the UI. In case the path contains a number, it means that the algorithm found a difference within an array, so we truncate it right there to obtain only the path to the array object. That way we can gather all the changes within the array under a bigger umbrella, like `List changes`. That's why we need the trucated path.

Reply:
That's super insightful and important information. Please put that in comments.